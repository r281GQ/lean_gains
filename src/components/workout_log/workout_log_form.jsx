import React from 'react';
import ExerciseFieldArray from './exercies_field_array';
import { Field, FieldArray } from 'redux-form/immutable';
import { RaisedButton, Paper } from 'material-ui';
import { DatePicker } from 'redux-form-material-ui';

import DateSelector from './date_selector';

const WorkoutLogForm = ({
  disabled,
  passedMarkerList,
  normalizeMarker,
  label,
  submitHandler,
  renderDate,
  maxDate,
  minDate,
  formatDate,
  shouldDisableDate,
  weightLabel
}) =>
  <div className="workout-log-container">
    <Paper className="workout-log-paper">
      <form onSubmit={submitHandler}>
        <div>
          <DateSelector
            renderDate={renderDate}
            shouldDisableDate={shouldDisableDate}
            maxDate={maxDate}
            minDate={minDate}
            formatDate={formatDate}
          />
          <FieldArray
            name="exercises"
            component={ExerciseFieldArray}
            passedMarkerList={passedMarkerList}
            normalizeMarker={normalizeMarker}
          />
          <RaisedButton
            fullWidth
            type="submit"
            disabled={disabled}
            label={label}
          />
        </div>
      </form>
    </Paper>
  </div>;

export default WorkoutLogForm;
