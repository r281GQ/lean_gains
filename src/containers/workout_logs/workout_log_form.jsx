import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ImmutablePropTypes from 'react-immutable-proptypes';
import { reduxForm, getFormValues } from 'redux-form/immutable';
import { Map, List } from 'immutable';
import moment from 'moment';
import * as _ from 'lodash';

import WorkoutLogForm from './../../components/workout_log/workout_log_form';
import {
  createWorkoutLog,
  updateWorkoutLog
} from './../../store/actionCreators/workout_log_action_creators';

// TODO: move mapStateToProps to selectors
class WorkoutLogFormContainer extends PureComponent {
  constructor(props) {
    super(props);
    this._disableThese = this._disableThese.bind(this);
    this._isDisabled = this._isDisabled.bind(this);
    this._handleFormSubmit = this._handleFormSubmit.bind(this);
    this._renderDate = this._renderDate.bind(this);
    this._mapToBoolean = this._mapToBoolean.bind(this);
  }

  componentDidMount() {
    if (this.props.match.params.id)
      this.props.initialize(
        this.props.defaultValue.update('createdAt', value =>
          moment(value).toDate()
        )
      );
    else {
      this.props.initialize(Map().set('createdAt', moment().toDate()));
    }
  }

  _mapToBoolean(value) {
    return _.isBoolean(value) ? value : false;
  }

  _disableThese(disableDates) {
    return function(createdAt) {
      return disableDates.find(value => moment(value).isSame(createdAt, 'day'))
        ? true
        : false;
    };
  }

  _isDisabled() {
    const { match, datesWithWorkoutLogs, selectedDate } = this.props;
    return datesWithWorkoutLogs.find(value =>
      moment(value).isSame(selectedDate, 'day')
    ) && !match.params.id
      ? true
      : false;
  }

  _handleFormSubmit(formProps) {
    const { match, createWorkoutLog, updateWorkoutLog } = this.props;
    match.params.id
      ? updateWorkoutLog({
          ...formProps.toJS(),
          _id: match.params.id,
          createdAt: moment(formProps.get('createdAt')).valueOf()
        })
      : createWorkoutLog({
          ...formProps.toJS(),
          createdAt: moment(formProps.get('createdAt')).valueOf()
        });
  }

  _renderDate() {
    const { match } = this.props;
    return match.path ? _.includes(match.path, 'before') : false;
  }

  render() {
    const {
      match,
      handleSubmit,
      datesWithWorkoutLogs,
      markerList
    } = this.props;
    return (
      <WorkoutLogForm
        submitHandler={handleSubmit(this._handleFormSubmit)}
        renderDate={this._renderDate()}
        maxDate={moment().toDate()}
        minDate={moment()
          .subtract(110, 'years')
          .toDate()}
        formatDate={value => moment(value).toDate()}
        shouldDisableDate={this._disableThese(datesWithWorkoutLogs)}
        passedMarkerList={markerList}
        normalizeMarker={this._mapToBoolean}
        disabled={this._isDisabled()}
        label={match.params.id ? 'Update' : 'Create'}
      />
    );
  }
}

WorkoutLogFormContainer.propTypes = {
  createWorkoutLog: PropTypes.func.isRequired,
  defaultValue: PropTypes.oneOfType([
    ImmutablePropTypes.list,
    ImmutablePropTypes.map
  ]),
  updateWorkoutLog: PropTypes.func.isRequired,
  datesWithWorkoutLogs: ImmutablePropTypes.set,
  markerList: ImmutablePropTypes.list,
  selectedDate: PropTypes.instanceOf(Date)
};

const mapStateToProps = state => {
  return {
    datesWithWorkoutLogs: state.getIn(['workoutLogs', 'dates']),
    selectedDate: getFormValues('workoutLog')(state)
      ? moment(getFormValues('workoutLog')(state).get('createdAt')).toDate()
      : moment().toDate(),
    markerList:
      getFormValues('workoutLog')(state) &&
      getFormValues('workoutLog')(state).get('exercises')
        ? getFormValues('workoutLog')(state)
            .get('exercises')
            .map(value => value.get('marker'))
        : List()
  };
};

export default connect(mapStateToProps, { updateWorkoutLog, createWorkoutLog })(
  reduxForm({
    form: 'workoutLog'
  })(WorkoutLogFormContainer)
);

export {WorkoutLogFormContainer as PureWorkoutLogFormContainer}
