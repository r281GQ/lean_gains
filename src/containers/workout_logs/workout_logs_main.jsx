import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import moment from 'moment';

import {
  setSelectedMonthForWorkoutLogs,
  openWorkoutModal,
  closeWorkoutModal,
  setSelectedWorkoutLog
} from './../../store/actionCreators/app_action_creators';
import {
  getWorkoutLogsForMonth,
  deleteWorkoutLog,
  getWorkoutLogDates
} from './../../store/actionCreators/workout_log_action_creators';
import monthsWithWorkoutLogs from './../../store/selectors/month_workout_log';
import workoutLogsForMonth from './../../store/selectors/workout_log_selector';
import isTodaysWorkoutLogExists from './../../store/selectors/today_log';

import CreateButton from './../../components/create_button';
import CreateButtonMinified from './../../components/create_button_minified';
import ConfirmDelete from './../../components/confirm_delete';
import CardListLog from './../../components/card_list_log';
import DateSelector from './../../components/date_selector';
import LoadingScreen from './../../components/loading';

const renderMainScreen = ({
  monthsWithWorkoutLogs,
  selectedMonth,
  getWorkoutLogsForMonth,
  setSelectedMonthForWorkoutLogs,
  isModalOpen,
  closeWorkoutModal,
  deleteWorkoutLog,
  workoutLogsForMonth,
  setSelectedWorkoutLog,
  openWorkoutModal,
  isTodaysWorkoutLogExists,
  selectedWorkoutLog
}) =>
  <div>
    <DateSelector
      months={monthsWithWorkoutLogs.toJS()}
      selectedMonth={selectedMonth}
      fetchDataForSelectedMonth={getWorkoutLogsForMonth}
      setSelectedMonth={setSelectedMonthForWorkoutLogs}
    />

    <ConfirmDelete
      title="Sure you want to delete this log?"
      isOpen={isModalOpen}
      close={closeWorkoutModal}
      deleteActions={[() => deleteWorkoutLog(selectedWorkoutLog)]}
    />

    <CardListLog
      workoutLogs={workoutLogsForMonth.toJS()}
      editLink="/workoutlogs/edit/"
      setSelectedItem={setSelectedWorkoutLog}
      onModalStateChange={openWorkoutModal}
    />

    <CreateButtonMinified link="/workoutlogs/create/before" />

    <CreateButton link="/workoutlogs/create" disabled={isTodaysWorkoutLogExists} />
  </div>;

class WorkoutLogsMainContainer extends PureComponent {
  componentWillMount = () => {
    if (this.props.datesWithWorkoutLogs.isEmpty())
      this.props.getWorkoutLogDates();
    if (this.props.workoutLogsForMonth.isEmpty())
      this.props.getWorkoutLogsForMonth(moment().format('MM-YYYY'));
    this.props.setSelectedMonthForWorkoutLogs(
      this.props.monthsWithWorkoutLogs.last()
    );
  };

  componentWillReceiveProps = (nextProps, nextState) =>
    this.props.monthsWithWorkoutLogs.isEmpty()
      ? this.props.setSelectedMonthForWorkoutLogs(
          nextProps.monthsWithWorkoutLogs.last()
        )
      : null;

  render = () =>
    this.props.isLoading ? <LoadingScreen /> : renderMainScreen(this.props);
}

const mapStateToProps = state => ({
  isLoading: state.getIn(['app', 'isLoading']),
  monthsWithWorkoutLogs: monthsWithWorkoutLogs(state),
  isTodaysWorkoutLogExists: isTodaysWorkoutLogExists(state),
  workoutLogsForMonth: workoutLogsForMonth(state),
  datesWithWorkoutLogs: state.getIn(['workoutLogs', 'dates']),
  selectedMonth: state.getIn(['app', 'selectedMonthForWorkoutLogs']),
  isModalOpen: state.getIn(['app', 'isWorkoutLogModalOpen']),
  selectedWorkoutLog: state.getIn(['app', 'selectedWorkoutLog'])
});

const mapDispatchToProps = dispatch => ({
  getWorkoutLogsForMonth: month => dispatch(getWorkoutLogsForMonth(month)),
  getWorkoutLogDates: () => dispatch(getWorkoutLogDates()),
  setSelectedMonthForWorkoutLogs: month =>
    dispatch(setSelectedMonthForWorkoutLogs(month)),
  deleteWorkoutLog: _id => dispatch(deleteWorkoutLog(_id)),
  openWorkoutModal: () => dispatch(openWorkoutModal()),
  closeWorkoutModal: () => dispatch(closeWorkoutModal()),
  setSelectedWorkoutLog: _id => dispatch(setSelectedWorkoutLog(_id))
});

export default connect(mapStateToProps, mapDispatchToProps)(
  WorkoutLogsMainContainer
);
