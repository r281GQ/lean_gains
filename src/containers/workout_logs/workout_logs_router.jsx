import React from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import { connect } from 'react-redux';
import { Route } from 'react-router-dom';
import { Map } from 'immutable';

import WorkoutLogsMainContainer from './workout_logs_main';
import WorkoutLogFormContainer from './workout_log_form';
import isTrainingDay from './../../store/selectors/exercises';

import withConfirmDeleteModal from './../enhancers/confirm_delete_modal';

const WorkoutLogsRouter = ({ exercises, workoutLogs }) =>
  <div>
    <Route
      exact
      path={'/app/workoutlogs'}
      component={withConfirmDeleteModal(
        WorkoutLogsMainContainer,
        `Are you sure you want to delete this workout log?`,
        'workoutLog',
      )}
    />

    <Route
      exact
      path={'/app/workoutlogs/create/before'}
      render={props => {
        return <WorkoutLogFormContainer {...props} type={`createBefore`} />;
      }}
    />

    <Route
      exact
      path="/app/workoutlogs/edit/:id"
      render={props => {
        const defaultValue = workoutLogs
          .find((value, key) => props.match.params.id === key)
          // .get('exercises');
;
        return (
          <WorkoutLogFormContainer
            {...props}
            defaultValue={defaultValue}

          />
        );
      }}
    />

    <Route
      exact
      path={'/app/workoutlogs/create'}
      render={props => {
        const defaultValue = exercises.map(value =>
          Map().withMutations(map =>
            map
              .set(
                'name',
                value !== 'You do not have any exercise for today' ? value : '',
              )
              .set('note', '')
              .set('marker', false)
              .set('sets', Map()),
          ),
        );

        return (
          <WorkoutLogFormContainer
            {...props}
            type={`create`}
            defaultValue={defaultValue}
          />
        );
      }}
    />
  </div>;

WorkoutLogsRouter.propTypes = {
  exercises: ImmutablePropTypes.list,
  workoutLogs: ImmutablePropTypes.map,
};

export default connect(state => ({
  workoutLogs: state.getIn(['workoutLogs', 'data']),
  exercises: isTrainingDay('main')(state),
}))(WorkoutLogsRouter);
