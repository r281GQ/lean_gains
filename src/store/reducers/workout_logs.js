import { fromJS, Map, Set } from 'immutable';
import * as _ from 'lodash';

import * as workoutLogs from './../actions/workout_logs_actions';

const INITIAL_STATE = Map().withMutations(map =>
  map.set('data', Map()).set('dates', Set())
);

const reducer = (state = INITIAL_STATE, { type, payload }) => {
  switch (type) {
    case workoutLogs.WRITE_WORKOUT_LOG_DATES:
      return state.update('dates', set => set.concat(fromJS(payload)));
    case workoutLogs.WRITE_WORKOUT_LOG_DATE:
      return state.update('dates', set => set.add(payload));
    case workoutLogs.DELETE_WORKOUT_LOG_DATE:
      return state.update('dates', set => set.remove(payload));
    case workoutLogs.DELETE_WORKOUT_LOG:
      return state.deleteIn(['data', payload]);
    case workoutLogs.WRITE_WORKOUT_LOGS:
      return state.update('data', map =>
        map.concat(fromJS(_.keyBy(payload, '_id')))
      );
    case workoutLogs.WRITE_WORKOUT_LOG:
      return state.setIn(['data', payload._id], fromJS(payload));
    default:
      return state;
  }
};

export default reducer;
