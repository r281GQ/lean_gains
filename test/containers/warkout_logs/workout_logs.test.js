import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import {Set} from 'immutable';
import moment from 'moment';
import {PureWorkoutLogFormContainer} from './../../../src/containers/workout_logs/workout_log_form';

describe('WorkoutLogs test', () => {
  it('should render', () => {
    const dateStub = sinon.stub(Date, 'now').returns(1);
    const updateWorkoutLogSpy = sinon.spy();
    const createWorkoutLogSpy = sinon.spy();


    const component = shallow(
      <PureWorkoutLogFormContainer

        handleSubmit ={() => ()=> undefined}
        updateWorkoutLog = {updateWorkoutLogSpy}
        createWorkoutLog = {createWorkoutLogSpy}
        match = {{params: {id: 1}}}
        datesWithWorkoutLogs = {Set()}
      />
    );
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
     dateStub.restore();
  });

  it('should render', () => {
    const dateStub = sinon.stub(Date, 'now').returns(1);
    const updateWorkoutLogSpy = sinon.spy();
    const createWorkoutLogSpy = sinon.spy();


    const component = shallow(
      <PureWorkoutLogFormContainer

        handleSubmit ={() => ()=> undefined}
        updateWorkoutLog = {updateWorkoutLogSpy}
        createWorkoutLog = {createWorkoutLogSpy}
        match = {{params: {id: 1}, path: 'bedsgfdfore'}}
        datesWithWorkoutLogs = {Set()}
      />
    );


    const inst = component.instance();

    expect(inst._renderDate()).toBe(false)

    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
     dateStub.restore();
  });

  it('should render', () => {
    const dateStub = sinon.stub(Date, 'now').returns(1);
    const updateWorkoutLogSpy = sinon.spy();
    const createWorkoutLogSpy = sinon.spy();


    const component = shallow(
      <PureWorkoutLogFormContainer

        handleSubmit ={() => ()=> undefined}
        updateWorkoutLog = {updateWorkoutLogSpy}
        createWorkoutLog = {createWorkoutLogSpy}
        match = {{params:{}}}
        datesWithWorkoutLogs = {Set()}
      />
    );


    const inst = component.instance();

    expect(inst._renderDate()).toBe(false)

    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
     dateStub.restore();
  });

  it('should render', () => {
    const dateStub = sinon.stub(Date, 'now').returns(1);
    const updateWorkoutLogSpy = sinon.spy();
    const createWorkoutLogSpy = sinon.spy();


    const component = shallow(
      <PureWorkoutLogFormContainer

        handleSubmit ={() => ()=> undefined}
        updateWorkoutLog = {updateWorkoutLogSpy}
        createWorkoutLog = {createWorkoutLogSpy}
        match = {{params: {id: 1}, path: 'before'}}
        datesWithWorkoutLogs = {Set()}
      />
    );


    const inst = component.instance();

    expect(inst._renderDate()).toBe(true)

    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
     dateStub.restore();
  });

  it('should render', () => {
    const dateStub = sinon.stub(Date, 'now').returns(1);
    const updateWorkoutLogSpy = sinon.spy();
    const createWorkoutLogSpy = sinon.spy();


    const component = shallow(
      <PureWorkoutLogFormContainer

        handleSubmit ={() => ()=> undefined}
        updateWorkoutLog = {updateWorkoutLogSpy}
        createWorkoutLog = {createWorkoutLogSpy}
        match = {{params: {id: 1}, path: 'before'}}
        datesWithWorkoutLogs = {Set()}
      />
    );


    const inst = component.instance();

    // expect(inst._renderDate()).toBe(true)
inst._handleFormSubmit(Set())
    expect(updateWorkoutLogSpy.calledOnce).toBe(true)
    expect(createWorkoutLogSpy.calledOnce).toBe(false)
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
     dateStub.restore();
  });

  it('should render', () => {
    const dateStub = sinon.stub(Date, 'now').returns(1);
    const updateWorkoutLogSpy = sinon.spy();
    const createWorkoutLogSpy = sinon.spy();


    const component = shallow(
      <PureWorkoutLogFormContainer

        handleSubmit ={() => ()=> undefined}
        updateWorkoutLog = {updateWorkoutLogSpy}
        createWorkoutLog = {createWorkoutLogSpy}
        match = {{params: {}, path: 'before'}}
        datesWithWorkoutLogs = {Set()}
      />
    );


    const inst = component.instance();

    // expect(inst._renderDate()).toBe(true)
inst._handleFormSubmit(Set())
    expect(updateWorkoutLogSpy.calledOnce).toBe(false)
    expect(createWorkoutLogSpy.calledOnce).toBe(true)
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
     dateStub.restore();
  });

  it('should render', () => {
    const dateStub = sinon.stub(Date, 'now').returns(1);
    const updateWorkoutLogSpy = sinon.spy();
    const createWorkoutLogSpy = sinon.spy();


    const component = shallow(
      <PureWorkoutLogFormContainer

        handleSubmit ={() => ()=> undefined}
        updateWorkoutLog = {updateWorkoutLogSpy}
        createWorkoutLog = {createWorkoutLogSpy}
        match = {{params: {}, path: 'before'}}
        datesWithWorkoutLogs = {Set()}
        selectedDate ={moment(10000).toDate()}
      />
    );


    const inst = component.instance();

    expect(inst._isDisabled()).toBe(false)
// inst._handleFormSubmit(Set())
    // expect(updateWorkoutLogSpy.calledOnce).toBe(false)
    // expect(createWorkoutLogSpy.calledOnce).toBe(true)
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
     dateStub.restore();
  });

  it('should render', () => {
    const dateStub = sinon.stub(Date, 'now').returns(1);
    const updateWorkoutLogSpy = sinon.spy();
    const createWorkoutLogSpy = sinon.spy();


    const component = shallow(
      <PureWorkoutLogFormContainer

        handleSubmit ={() => ()=> undefined}
        updateWorkoutLog = {updateWorkoutLogSpy}
        createWorkoutLog = {createWorkoutLogSpy}
        match = {{params: {id: 1}, path: 'before'}}
        datesWithWorkoutLogs = {Set().add(moment(1).toDate())}
        selectedDate ={moment(1).toDate()}
      />
    );


    const inst = component.instance();

    expect(inst._isDisabled()).toBe(false)
// inst._handleFormSubmit(Set())
    // expect(updateWorkoutLogSpy.calledOnce).toBe(false)
    // expect(createWorkoutLogSpy.calledOnce).toBe(true)
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
     dateStub.restore();
  });

  it('should render', () => {
    const dateStub = sinon.stub(Date, 'now').returns(1);
    const updateWorkoutLogSpy = sinon.spy();
    const createWorkoutLogSpy = sinon.spy();


    const component = shallow(
      <PureWorkoutLogFormContainer

        handleSubmit ={() => ()=> undefined}
        updateWorkoutLog = {updateWorkoutLogSpy}
        createWorkoutLog = {createWorkoutLogSpy}
        match = {{params: {}, path: 'before'}}
        datesWithWorkoutLogs = {Set().add(moment(1).toDate())}
        selectedDate ={moment(1).toDate()}
      />
    );


    const inst = component.instance();

    expect(inst._isDisabled()).toBe(true)
// inst._handleFormSubmit(Set())
    // expect(updateWorkoutLogSpy.calledOnce).toBe(false)
    // expect(createWorkoutLogSpy.calledOnce).toBe(true)
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
     dateStub.restore();
  });

  it('should render', () => {
    const dateStub = sinon.stub(Date, 'now').returns(1);
    const updateWorkoutLogSpy = sinon.spy();
    const createWorkoutLogSpy = sinon.spy();


    const component = shallow(
      <PureWorkoutLogFormContainer

        handleSubmit ={() => ()=> undefined}
        updateWorkoutLog = {updateWorkoutLogSpy}
        createWorkoutLog = {createWorkoutLogSpy}
        match = {{params: {}, path: 'before'}}
        datesWithWorkoutLogs = {Set().add(moment(1).toDate())}
        selectedDate ={moment(1).toDate()}
      />
    );


    const inst = component.instance();
    const t = inst._disableThese(Set().add(moment('01-01-2017', 'DD-MM-YYYY').toDate()))
    expect(t(moment('01-01-2017', 'DD-MM-YYYY').toDate())).toBe(true)
// inst._handleFormSubmit(Set())
    // expect(updateWorkoutLogSpy.calledOnce).toBe(false)
    // expect(createWorkoutLogSpy.calledOnce).toBe(true)
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
     dateStub.restore();
  });

  it('should render', () => {
    const dateStub = sinon.stub(Date, 'now').returns(1);
    const updateWorkoutLogSpy = sinon.spy();
    const createWorkoutLogSpy = sinon.spy();


    const component = shallow(
      <PureWorkoutLogFormContainer

        handleSubmit ={() => ()=> undefined}
        updateWorkoutLog = {updateWorkoutLogSpy}
        createWorkoutLog = {createWorkoutLogSpy}
        match = {{params: {}, path: 'before'}}
        datesWithWorkoutLogs = {Set().add(moment(1).toDate())}
        selectedDate ={moment(1).toDate()}
      />
    );


    const inst = component.instance();
    const t = inst._disableThese(Set().add(moment('01-01-2017', 'DD-MM-YYYY').toDate()))
    expect(t(moment('01-02-2017', 'DD-MM-YYYY').toDate())).toBe(false)
// inst._handleFormSubmit(Set())
    // expect(updateWorkoutLogSpy.calledOnce).toBe(false)
    // expect(createWorkoutLogSpy.calledOnce).toBe(true)
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
     dateStub.restore();
  });
});
