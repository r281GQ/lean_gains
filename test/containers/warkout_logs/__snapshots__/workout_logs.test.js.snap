// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkoutLogs test should render 1`] = `
<WorkoutLogForm
  disabled={false}
  formatDate={[Function]}
  label="Update"
  maxDate={1970-01-01T00:00:00.001Z}
  minDate={1860-01-01T00:00:00.001Z}
  normalizeMarker={[Function]}
  renderDate={false}
  shouldDisableDate={[Function]}
  submitHandler={[Function]}
/>
`;

exports[`WorkoutLogs test should render 2`] = `
<WorkoutLogForm
  disabled={false}
  formatDate={[Function]}
  label="Update"
  maxDate={1970-01-01T00:00:00.001Z}
  minDate={1860-01-01T00:00:00.001Z}
  normalizeMarker={[Function]}
  renderDate={false}
  shouldDisableDate={[Function]}
  submitHandler={[Function]}
/>
`;

exports[`WorkoutLogs test should render 3`] = `
<WorkoutLogForm
  disabled={false}
  formatDate={[Function]}
  label="Create"
  maxDate={1970-01-01T00:00:00.001Z}
  minDate={1860-01-01T00:00:00.001Z}
  normalizeMarker={[Function]}
  renderDate={false}
  shouldDisableDate={[Function]}
  submitHandler={[Function]}
/>
`;

exports[`WorkoutLogs test should render 4`] = `
<WorkoutLogForm
  disabled={false}
  formatDate={[Function]}
  label="Update"
  maxDate={1970-01-01T00:00:00.001Z}
  minDate={1860-01-01T00:00:00.001Z}
  normalizeMarker={[Function]}
  renderDate={true}
  shouldDisableDate={[Function]}
  submitHandler={[Function]}
/>
`;

exports[`WorkoutLogs test should render 5`] = `
<WorkoutLogForm
  disabled={false}
  formatDate={[Function]}
  label="Update"
  maxDate={1970-01-01T00:00:00.001Z}
  minDate={1860-01-01T00:00:00.001Z}
  normalizeMarker={[Function]}
  renderDate={true}
  shouldDisableDate={[Function]}
  submitHandler={[Function]}
/>
`;

exports[`WorkoutLogs test should render 6`] = `
<WorkoutLogForm
  disabled={false}
  formatDate={[Function]}
  label="Create"
  maxDate={1970-01-01T00:00:00.001Z}
  minDate={1860-01-01T00:00:00.001Z}
  normalizeMarker={[Function]}
  renderDate={true}
  shouldDisableDate={[Function]}
  submitHandler={[Function]}
/>
`;

exports[`WorkoutLogs test should render 7`] = `
<WorkoutLogForm
  disabled={false}
  formatDate={[Function]}
  label="Create"
  maxDate={1970-01-01T00:00:00.001Z}
  minDate={1860-01-01T00:00:00.001Z}
  normalizeMarker={[Function]}
  renderDate={true}
  shouldDisableDate={[Function]}
  submitHandler={[Function]}
/>
`;

exports[`WorkoutLogs test should render 8`] = `
<WorkoutLogForm
  disabled={false}
  formatDate={[Function]}
  label="Update"
  maxDate={1970-01-01T00:00:00.001Z}
  minDate={1860-01-01T00:00:00.001Z}
  normalizeMarker={[Function]}
  renderDate={true}
  shouldDisableDate={[Function]}
  submitHandler={[Function]}
/>
`;

exports[`WorkoutLogs test should render 9`] = `
<WorkoutLogForm
  disabled={true}
  formatDate={[Function]}
  label="Create"
  maxDate={1970-01-01T00:00:00.001Z}
  minDate={1860-01-01T00:00:00.001Z}
  normalizeMarker={[Function]}
  renderDate={true}
  shouldDisableDate={[Function]}
  submitHandler={[Function]}
/>
`;

exports[`WorkoutLogs test should render 10`] = `
<WorkoutLogForm
  disabled={true}
  formatDate={[Function]}
  label="Create"
  maxDate={1970-01-01T00:00:00.001Z}
  minDate={1860-01-01T00:00:00.001Z}
  normalizeMarker={[Function]}
  renderDate={true}
  shouldDisableDate={[Function]}
  submitHandler={[Function]}
/>
`;

exports[`WorkoutLogs test should render 11`] = `
<WorkoutLogForm
  disabled={true}
  formatDate={[Function]}
  label="Create"
  maxDate={1970-01-01T00:00:00.001Z}
  minDate={1860-01-01T00:00:00.001Z}
  normalizeMarker={[Function]}
  renderDate={true}
  shouldDisableDate={[Function]}
  submitHandler={[Function]}
/>
`;
